common/profanity_manager.cpp

```diff
@@ -58,83 +60,122 @@ bool EQ::ProfanityManager::DeleteProfanityList(DBcore *db) {
 	return true;
 }
 
-bool EQ::ProfanityManager::AddProfanity(DBcore *db, const char *profanity) {
-	if (!db || !profanity)
+bool EQ::ProfanityManager::AddProfanity(DBcore *db, std::string profanity) {
+	if (!db || profanity.empty()) {
 		return false;
+	}
 
 	std::string entry(profanity);
 
-	std::transform(entry.begin(), entry.end(), entry.begin(), [](unsigned char c) -> unsigned char { return tolower(c); });
+	std::transform(
+		entry.begin(),
+		entry.end(),
+		entry.begin(),
+		[](unsigned char c) -> unsigned char {
+			return tolower(c);
+		}
+	);
 
-	if (check_for_existing_entry(entry.c_str()))
+	if (check_for_existing_entry(entry)) {
 		return true;
+	}
 
-	if (entry.length() < REDACTION_LENGTH_MIN)
+	if (entry.length() < REDACTION_LENGTH_MIN) {
 		return false;
+	}
 
 	profanity_list.push_back(entry);
 
-	std::string query = "REPLACE INTO `profanity_list` (`word`) VALUES ('";
-	query.append(entry);
-	query.append("')");
+	auto query = fmt::format(
+		"REPLACE INTO `profanity_list` (`word`) VALUES ('{}')",
+		profanity
+	);
 	auto results = db->QueryDatabase(query);
-	if (!results.Success())
+
+	if (!results.Success()) {
 		return false;
+	}
 
 	update_originator_flag = true;
 
 	return true;
 }
 
-bool EQ::ProfanityManager::RemoveProfanity(DBcore *db, const char *profanity) {
-	if (!db || !profanity)
+bool EQ::ProfanityManager::RemoveProfanity(DBcore *db, std::string profanity) {
+	if (!db || profanity.empty()) {
 		return false;
+	}
 
 	std::string entry(profanity);
 
-	std::transform(entry.begin(), entry.end(), entry.begin(), [](unsigned char c) -> unsigned char { return tolower(c); });
+	std::transform(
+		entry.begin(),
+		entry.end(),
+		entry.begin(),
+		[](unsigned char c) -> unsigned char {
+			return tolower(c);
+		}
+	);
 
-	if (!check_for_existing_entry(entry.c_str()))
+	if (!check_for_existing_entry(entry)) {
 		return true;
+	}
 
 	profanity_list.remove(entry);
 
-	std::string query = "DELETE FROM `profanity_list` WHERE `word` LIKE '";
-	query.append(entry);
-	query.append("'");
+	auto query = fmt::format(
+		"DELETE FROM `profanity_list` WHERE `word` = '{}'",
+		entry
+	);
 	auto results = db->QueryDatabase(query);
-	if (!results.Success())
+
+	if (!results.Success()) {
 		return false;
+	}
 
 	update_originator_flag = true;
 
 	return true;
 }
 
 void EQ::ProfanityManager::RedactMessage(char *message) {
-	if (!message)
+	if (!message) {
 		return;
+	}
 
 	std::string test_message(message);
 	// hard-coded max length based on channel message buffer size (4096 bytes)..
 	// ..will need to change or remove if other sources are used for redaction
-	if (test_message.length() < REDACTION_LENGTH_MIN || test_message.length() >= 4096)
+	if (test_message.length() < REDACTION_LENGTH_MIN || test_message.length() >= 4096) {
 		return;
+	}
 
-	std::transform(test_message.begin(), test_message.end(), test_message.begin(), [](unsigned char c) -> unsigned char { return tolower(c); });
+	std::transform(
```

Comment:
We have utilities to cover this in `string_util.h` `to_lower`

I realize this isn't your code but if we're cleaning up